### PR TITLE
Rust TDD Features

### PR DESCRIPTION
**Pull request by droogmic**
Mainly reworking Bicycle, Sidewalk and Shoulder

### CODE REVIEW

File: rust/osm2lanes/src/lib.rs [hunk 1]
```diff
@@ -316,42 +320,38 @@ mod tests {
...
-                } else if test.link.is_some() {
-                    println!("For input (example from {}):", test.link.as_ref().unwrap());
-                }
-                println!("From:");
-                println!("    {}", stringify_lane_types(&input_road));
-                println!("    {}", stringify_directions(&input_road));
-                println!("Normalized OSM tags:");
-                for (k, v) in tags.map() {
-                    println!("    {} = {}", k, v);
+        assert!(
```
Comment (droogmic) [inline rust/osm2lanes/src/lib.rs:323]: that is a very unfortunate rust fmt :/

File: kotlin/build.gradle.kts [hunk 2]
```diff
@@ -17,6 +17,7 @@ dependencies {
 
 tasks.test {
     useJUnit()
+    inputs.files(file("data/tests.json"))
```
Comment (droogmic) [inline kotlin/build.gradle.kts:20]: I don't know what this should be, but I had issues where kotlin wasn't rerunning when the json changed...

Comment (dabreegster) [reply kotlin/build.gradle.kts:20]: @enzet happen to know?

File: rust/osm2lanes/src/transform/tags_to_lanes.rs [hunk 3]
```diff
@@ -499,83 +526,163 @@ fn parking(
...
-        forward_side.push(Lane::foot());
-        backward_side.push(Lane::foot());
-    } else if tags.is("sidewalk", "separate") && locale.infer_sidewalks {
-        // TODO Need to snap separate sidewalks to ways. Until then, just do this.
-        forward_side.push(Lane::foot());
-        if !backward_side.is_empty() {
-            backward_side.push(Lane::foot());
+    warnings: &mut LaneWarnings,
+) -> ModeResult {
+    // https://wiki.openstreetmap.org/wiki/Key:sidewalk
```
Comment (dabreegster) [inline rust/osm2lanes/src/transform/tags_to_lanes.rs:537]: There were so many bizarre assumptions baked in here from old A/B Street design choices. Nice to see it done from scratch properly!

**Review by dabreegster — APPROVED** [review]

File: kotlin/build.gradle.kts (hunk 2 above)
Comment (droogmic) [reply kotlin/build.gradle.kts:20]: I will merge, to avoid being blocked. (but I still want to know how to fix this) :)
I never liked the gradle DSL...

**Pull request merged**